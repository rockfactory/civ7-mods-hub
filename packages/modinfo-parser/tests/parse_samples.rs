use modinfo_parser::Mod;

#[test]
fn parse_samples() {
    insta::glob!("samples/*.modinfo", |path| {
        let text = std::fs::read_to_string(path).unwrap();
        let parsed = Mod::parse(text.as_bytes()).unwrap();
        insta::with_settings!({ sort_maps => true }, {
            insta::assert_debug_snapshot!(parsed);
            let serialized = parsed.to_string().unwrap();
            let reparsed = Mod::parse(serialized.as_bytes()).unwrap();
            assert_eq!(parsed, reparsed);
        });
    });
}
