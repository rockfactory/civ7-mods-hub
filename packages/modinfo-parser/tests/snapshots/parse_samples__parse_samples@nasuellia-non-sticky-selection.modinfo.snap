---
source: packages/modinfo-parser/tests/parse_samples.rs
expression: parsed
input_file: packages/modinfo-parser/tests/samples/nasuellia-non-sticky-selection.modinfo
---
Mod {
    id: "nasuellia-non-sticky-selection",
    version: Some(
        "9",
    ),
    properties: Properties {
        name: Some(
            "nasuellia-non-sticky-selection",
        ),
        description: Some(
            "implements non-sticky selection",
        ),
        authors: Some(
            "nasuellia",
        ),
        package: Some(
            "Mod",
        ),
        affects_saved_games: false,
        special_thanks: None,
        version: None,
        compatibility: None,
        url: None,
    },
    dependencies: [
        Dependency {
            id: "core",
            title: Some(
                "LOC_MODULE_CORE_NAME",
            ),
        },
    ],
    localized_text: [],
    action_criteria: [
        ActionCriterion {
            id: "always",
            always_met: true,
            age_in_use: None,
            mod_in_use: None,
        },
    ],
    action_groups: [
        ActionGroup {
            id: "nasuellia-non-sticky-selection-scripts",
            scope: Some(
                "game",
            ),
            criteria: Some(
                "always",
            ),
            properties: ActionGroupProperties {
                load_order: Some(
                    101,
                ),
            },
            actions: [
                ActionGroupActions {
                    import_files: [
                        ActionItem {
                            path: "core/ui/input/action-handler.js",
                            locale: None,
                        },
                    ],
                    ui_scripts: [],
                    update_database: [],
                    update_text: [],
                },
            ],
        },
    ],
}
